### ISSUE TITLE
Additional analysis for paper

### ISSUE THREAD

**Issue by stevehadd** [MetOffice/XBTs_classification#103]
Based on discussion with Francesco, here are some additional things we could try for additional analysis to add to the paper to make a stronger case for how we have chosen to do things:
* cal;culate permutation importance to show which features are useful
* include balanced accuracy and some other metrics that are better suited to an imbalanced class problem than recall
* calculate some confusion matrices to shed more light on performance for different classes
* do some bootstrapping to get better performance for small classes through oversampling in the train set.

**Comment by stevehadd** [comment]
scikit learn bootstapping as part of cross validation:
https://ogrisel.github.io/scikit-learn.org/sklearn-tutorial/modules/generated/sklearn.cross_validation.Bootstrap.html

alternatively, the pandas sample method could be used for bootstrapping. 
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sample.html

We could use this to even out class support for a variety of features:
* instrument type
* country
* year

**Comment by stevehadd** [comment]
Some more ways to look at links between features
https://datascience.stackexchange.com/questions/893/how-to-get-correlation-between-two-categorical-variable-and-a-categorical-variab

**Comment by stevehadd** [comment]
various ways to do feature selection more systematically using sklearn or pandas:
* https://towardsdatascience.com/feature-selection-with-pandas-e3690ad8504b
* https://scikit-learn.org/stable/modules/feature_selection.html

**Comment by stevehadd** [comment]
Permutation feature importance algorithm implementation:
https://scikit-learn.org/stable/modules/permutation_importance.html

**Comment by stevehadd** [comment]
Docs for more metrics:
* https://scikit-learn.org/stable/modules/generated/sklearn.metrics.balanced_accuracy_score.html 
* https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html

**Comment by stevehadd** [comment]
This has been implemeted in various notebooks and updated for the batch code in PR #111.
